crates/ruff/src/autofix/source_map.rs

```diff
@@ -0,0 +1,59 @@
+use ruff_text_size::TextSize;
+
+use ruff_diagnostics::Edit;
+
+/// Lightweight sourcemap marker representing the source and destination
+/// position for an [`Edit`].
+#[derive(Debug, PartialEq, Eq)]
+pub struct SourceMarker {
```

Comment:
I think many of the `pub`s in here can be `pub(crate)`.

Reply:
Hmm, so it is possible it's just that the some parts will be public to allow `ruff_cli` to access.